controllers/main.js

```diff
@@ -180,20 +189,19 @@ module.exports.updatePlayData = async (req, res, next) => {
 
   if (thisUpdatedSession.length === parseInt(req.body.pageNum)) {
     const userId = thisUpdatedSession.userId;
-    const sessionDifficulty = remapDifficultyScoreKey[thisSession.difficulty];
     const currentUser = await User.findOne({ _id: userId });
     const score = calculateScore(thisUpdatedSession, Date.now());
     await PlaySession.updateOne(
       { sessionId },
       { $set: { sessionEnded: true, sessionScore: score } }
     );
     if (
-      !currentUser.bestScores[sessionDifficulty] ||
-      currentUser.bestScores[sessionDifficulty] < score
+      currentUser.bestScores[thisSession.difficulty] === undefined ||
```

Comment:
Can really just check `if(currentUser.bestScores[thisSession.difficulty] || currentUser.bestScores[thisSession.difficulty] < score`

Reply:
Is what I initially did as well but it did not seem to work, so I changed this part of the code to what it is now. 

Just in case tried out your suggested solution once again, in case i was doing something wrong previously myself, but it still does not seem to work.